File: frontend/src/api/modules/command.ts
/**
 * 控制命令分级确认 API
 * Story 9-6: 控制命令分级确认
 */
import request from '@/utils/request'
import type { PageParams, PageResponse } from './types'

// ==================== 类型定义 ====================

/** 命令提交请求 */
export interface CommandSubmitRequest {
  command_type: string
  target_device_id: number
  target_device_name: string
  command_content: Record<string, unknown>
}

/** 命令提交响应 */
export interface CommandSubmitResponse {
  status: string // executed / pending_approval
  message: string
  approval_id?: number
  audit_log_id?: number
}

/** 审批工单 */
export interface CommandApproval {
  id: number
  command_type: string
  risk_level: string
  target_device_id: number
  target_device_name: string
  command_content: Record<string, unknown> | null
  requester_id: number
  requester_name: string
  approver_id: number | null
  approver_name: string | null
  status: string
  reject_reason: string | null
  timeout_minutes: number
  created_at: string
  approved_at: string | null
  executed_at: string | null
  expired_at: string
}

/** 审计日志 */
export interface CommandAuditLog {
  id: number
  command_type: string
  risk_level: string
  target_device_id: number
  target_device_name: string
  command_content: Record<string, unknown> | null
  operator_id: number
  operator_name: string
  approval_id: number | null
  result: string
  result_message: string | null
  created_at: string
}

/** 风险配置项 */
export interface RiskConfigItem {
  command_type: string
  risk_level: 'normal' | 'critical'
  minimum_risk: 'normal' | 'critical'
  description?: string
}

// ==================== API 函数 ====================

/**
 * 提交控制命令
 */
export function submitCommand(data: CommandSubmitRequest): Promise<CommandSubmitResponse> {
  return request.post('/v1/command/submit', data)
}

/**
 * 获取审批工单列表
 */
export function getCommandApprovals(params?: PageParams & {
  status?: string
  requester_name?: string
}): Promise<PageResponse<CommandApproval>> {
  return request.get('/v1/command/approvals', { params })
}

/**
 * 获取审批工单详情
 */
export function getCommandApproval(id: number): Promise<CommandApproval> {
  return request.get(`/v1/command/approvals/${id}`)
}

/**
 * 批准审批
 */
export function approveCommand(id: number): Promise<CommandApproval> {
  return request.post(`/v1/command/approvals/${id}/approve`)
}

/**
 * 驳回审批
 */
export function rejectCommand(id: number, reason: string): Promise<CommandApproval> {
  return request.post(`/v1/command/approvals/${id}/reject`, { reason })
}

/**
 * 获取审计日志列表
 */
export function getCommandAuditLogs(params?: PageParams & {
  command_type?: string
  operator_name?: string
  result?: string
}): Promise<PageResponse<CommandAuditLog>> {
  return request.get('/v1/command/audit-logs', { params })
}

/**
 * 获取风险等级配置
 */
export function getRiskConfigs(): Promise<RiskConfigItem[]> {
  return request.get('/v1/command/risk-configs')
}

/**
 * 更新风险等级配置
 */
export function updateRiskConfigs(configs: RiskConfigItem[]): Promise<{ message: string; updated: number }> {
  return request.put('/v1/command/risk-configs', { configs })
}

File: backend/app/schemas/command.py
"""
控制命令分级确认 Schema
Story 9-6: 控制命令分级确认
"""

from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel, ConfigDict


# ==================== 命令提交 ====================


class CommandSubmitRequest(BaseModel):
    """提交控制命令请求"""

    command_type: str  # 命令类型标识: ac_temp_set, power_off 等
    target_device_id: int
    target_device_name: str
    command_content: dict  # 命令参数


class CommandSubmitResponse(BaseModel):
    """提交控制命令响应"""

    status: str  # executed / pending_approval
    message: str
    approval_id: Optional[int] = None
    audit_log_id: Optional[int] = None


# ==================== 审批工单 ====================


class CommandApprovalResponse(BaseModel):
    """审批工单响应"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    command_type: str
    risk_level: str
    target_device_id: int
    target_device_name: str
    command_content: Optional[dict] = None
    requester_id: int
    requester_name: str
    approver_id: Optional[int] = None
    approver_name: Optional[str] = None
    status: str
    reject_reason: Optional[str] = None
    timeout_minutes: int
    created_at: datetime
    approved_at: Optional[datetime] = None
    executed_at: Optional[datetime] = None
    expired_at: datetime


class ApprovalRejectRequest(BaseModel):
    """驳回审批请求"""

    reason: str


# ==================== 审计日志 ====================


class CommandAuditLogResponse(BaseModel):
    """命令审计日志响应"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    command_type: str
    risk_level: str
    target_device_id: int
    target_device_name: str
    command_content: Optional[dict] = None
    operator_id: int
    operator_name: str
    approval_id: Optional[int] = None
    result: str
    result_message: Optional[str] = None
    created_at: datetime


# ==================== 风险配置 ====================


class RiskConfigItem(BaseModel):
    """单条风险配置"""

    command_type: str
    risk_level: str  # normal / critical
    minimum_risk: Optional[str] = None
    description: Optional[str] = None


class RiskConfigUpdateRequest(BaseModel):
    """批量更新风险配置请求"""

    configs: List[RiskConfigItem]
